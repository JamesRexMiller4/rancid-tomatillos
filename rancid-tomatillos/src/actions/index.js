export const addMovies = movies => {
  return { 
    type: 'ADD_MOVIES',
    movies
  }
}

export const addUser = id => {
  return {
    type: 'ADD_USER',
    id
  }
}

export const updateLoggedIn = isLoggedIn => {
  return {
    type: 'UPDATE_LOGGED_IN',
    isLoggedIn: !isLoggedIn
  }
}

export const removeUser = () => {
  return {
    type: 'REMOVE_USER',
    user: {}
  }
}