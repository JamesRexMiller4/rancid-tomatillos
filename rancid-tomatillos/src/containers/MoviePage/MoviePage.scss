.movie-page-main {
  width: 100%;
  height: 70vh;
  overflow: auto;
}

.movie-page-section {
  width: 100%;
  height: 100vh;
  background-size: cover;
  display: flex;
  align-items: center;
  justify-content: space-around;
}

.movie-page-poster {
  width: 500px;
  height: 750px;
  background-size: cover;
  border: 4px solid black;
  border-radius: 10px;
}

.movie-info-container {
  width: 50%;
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
}

.movie-page-details {
  width: 100%;
  border: 4px solid black;
  background-color: rgba(0, 0, 0, .8);
  color: white;
  text-align: center;
  font-size: 1.5em;
  padding: 20px;
  border-radius: 10px;
}

.movie-overview {
  text-align: left;
}
