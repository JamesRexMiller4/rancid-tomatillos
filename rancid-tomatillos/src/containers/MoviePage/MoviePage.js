import React, { Component } from 'react';
import './MoviePage.scss';
import StarRatingComponent from 'react-star-rating-component';
import { connect } from 'react-redux';
import { fetchRatings, postUserRating, deleteApiRating } from '../../apiCalls';
import { updateUserRatings, deleteUserRating } from '../../actions';
import images from '../../images/images';

export class MoviePage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      rating: 1
    }
  }

  onStarClick = (nextValue) => {
    this.setState({rating: nextValue});
  }

  deleteRating = (userRatings, movieId) => {
    const movieRating = userRatings.find(userRating => userRating.movie_id === parseInt(movieId));
    const options = {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json'
      }

    }
    deleteApiRating(options, movieRating)
      .then(data => this.props.deleteUserRating(movieRating.id))
      .catch(err => console.log(err))
  }

  findMovieRating = (id, userRatings) => {
    const movieRating = userRatings.filter(userRating => userRating.movie_id === parseInt(id));
    if (movieRating.length) {
      const icon = this.determineIcon(movieRating[0].rating);
      return (
        <section className='average-rating-container'>
          <h2>User Rating</h2>
          <div className='rating-section'>
            <img src={icon} alt='rancid tomatillo' className='tomatillo-icon' />
            <h2>{movieRating[0].rating}</h2>
            <button className='delete-rating-btn' onClick={() => this.deleteRating(userRatings, id)} type=
            'button'>Delete Rating</button>
          </div>
        </section>
      )
    } else {
      const icon = this.determineIcon(this.state.rating);
      return (
        <section className='user-rating-container'>
          <h2>Rate This Movie</h2>
          <section>
            <StarRatingComponent
                name="rating"
                starCount={10}
                value={this.state.rating}
                onStarClick={this.onStarClick}
            />
          </section>
          <button className='rate-btn' onClick={this.handleRatingSubmit} 
          type='button'>{<img src={icon} alt='image of rancid tomatillo' className='tomatillo-icon' />}{this.state.rating} Rate It</button>
        </section>)
    }
  }

  determineIcon = (rating) => {
    if (rating < 4) {
      return images.badTomatillo;
    } else if (rating > 3 && rating < 7) {
      return images.okayTomatillo;
    } else if (rating > 6) {
      return images.goodTomatillo;
    }
  }

  handleRatingSubmit = () => {
    const { id, userId } = this.props;
    const options = {
      method: 'POST',
      body: JSON.stringify({
        movie_id: parseInt(id),
        rating: parseInt(this.state.rating)
      }),
      headers: {
      'Content-Type': 'application/json'
      }
    }

    postUserRating(options, userId)
      .then(data => {
        fetchRatings(userId)
          .then(data => this.props.updateUserRatings(data.ratings))
      })
      .catch(err => console.log(err))
  }

  render() {
    const { id, movies, isLoggedIn, userRatings } = this.props
    let movie = movies.movies.find(movie => movie.id === parseInt(id));
    const { average_rating, backdrop_path, overview, poster_path, release_date, title } = movie;
    let roundedAvgRating = Math.ceil(average_rating);
    const rating = this.findMovieRating(id, userRatings);
    const icon = this.determineIcon(roundedAvgRating)
    return (
      <main className='movie-page-main'>
        <section className='movie-page-section' style={{backgroundImage:`url(${backdrop_path})`}} >
          <article className='movie-page-poster' style={{backgroundImage: `url(${poster_path})`}}>
          </article>
            <section className='movie-info-container'>
              <article className='movie-page-details'>
                <h2>{title}</h2>
                <p className='movie-overview'>{overview}</p>
                <h3>Release Date</h3>
                <p>{release_date}</p>
              </article>
              <article className='movie-page-ratings'>
                <section className='average-rating-container'>
                  <h2>Average Rating</h2>
                  <div className='rating-section'>
                    <img src={icon} alt='rancid tomatillo' className='tomatillo-icon' />
                    <h2>{roundedAvgRating}</h2>
                  </div>
                </section>
                {isLoggedIn && rating}
              </article>
            </section>
        </section>
      </main>
    )
  }
}

export const mapDispatchToProps = dispatch => ({
  updateUserRatings: ratings => dispatch(updateUserRatings(ratings)),
  deleteUserRating: ratingId => dispatch(deleteUserRating(ratingId))
})

export const mapStateToProps = state => ({
  movies: state.movies,
  isLoggedIn: state.isLoggedIn,
  userId: state.userId,
  userRatings: state.userRatings
})

export default connect(mapStateToProps, mapDispatchToProps)(MoviePage);
