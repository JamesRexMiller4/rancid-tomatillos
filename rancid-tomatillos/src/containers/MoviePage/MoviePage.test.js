import React from 'react';
import {shallow} from 'enzyme';
import { MoviePage, mapStateToProps, mapDispatchToProps } from './MoviePage';
import { updateUserRatings } from '../../actions';

 
describe('Movie Page', () => {
  describe('Movie Page Component', () => {
    let wrapper, movies, userId, userRatings, updateUserRatings;
    beforeEach(() => {
      movies = {movies: 
        [{
          id:2,
          title:"Jumanji: The Next Level",
          poster_path:"https://image.tmdb.org/t/p/original//l4iknLOenijaB85Zyb5SxH1gGz8.jpg",
          backdrop_path:"https://image.tmdb.org/t/p/original//zTxHf9iIOCqRbxvl8W5QYKrsMLq.jpg",
          release_date:"2019-12-04",
          overview:"In Jumanji: The Next Level, the gang is back to escape the world's most dangerous game.",
          average_rating:6.166666666666667
        },
        {
          id:15,
          title:"Ad Astra",
          poster_path:"https://image.tmdb.org/t/p/original//xBHvZcjRiWyobQ9kxBhO6B2dtRI.jpg",
          backdrop_path:"https://image.tmdb.org/t/p/original//5BwqwxMEjeFtdknRV792Svo0K1v.jpg",
          release_date:"2019-09-17",
          overview:"The near future, astronaut Roy McBride undertakes a mission in search of the unknown.",
          average_rating:5
        },
        {
          id:16,
          title:"Star Wars: The Rise Of Skywalker",
          poster_path:"https://image.tmdb.org/t/p/original//xBHvZcjRiWyobQ9kxBhO6B2dtRI.jpg",
          backdrop_path:"https://image.tmdb.org/t/p/original//5BwqwxMEjeFtdknRV792Svo0K1v.jpg",
          release_date:"2019-09-17",
          overview:"The conclusion of the Skywalker timeline, in a galaxy far far away.",
          average_rating:5
        },
      
        {
          id:18,
          title:"The Count of Monte Cristo",
          poster_path:"https://image.tmdb.org/t/p/original//xBHvZcjRiWyobQ9kxBhO6B2dtRI.jpg",
          backdrop_path:"https://image.tmdb.org/t/p/original//5BwqwxMEjeFtdknRV792Svo0K1v.jpg",
          release_date:"2019-09-17",
          overview:"The ultimate tale of revenge and adventure.",
          average_rating:10
        }]
      };
      userRatings = [
        {
          id:22,
          user_id:6,
          movie_id:4,
          rating:10,
          created_at:"2019-12-19T21:47:31.647Z",
          updated_at:"2019-12-19T21:47:31.647Z"
        },
        {
          id:23,
          user_id:6,
          movie_id:15,
          rating:8,
          created_at:"2019-12-19T22:05:16.098Z",
          updated_at:"2019-12-19T22:05:16.098Z"
        },
        {
          id:16,
          user_id:6,
          movie_id:16,
          rating:5,
          created_at:"2019-12-19T22:05:16.098Z",
          updated_at:"2019-12-19T22:05:16.098Z"
        },
        {
          id:18,
          user_id:6,
          movie_id:18,
          rating:2,
          created_at:"2019-12-19T22:05:16.098Z",
          updated_at:"2019-12-19T22:05:16.098Z"
        }
      ]
        userId = 6;
        updateUserRatings = jest.fn();
    });
    
    it('should match the snapshot when the user is logged in and the movie is not rated', () => {
      const isLoggedIn = true
      wrapper = shallow(<MoviePage 
        id={2}
        movies={movies}
        isLoggedIn={isLoggedIn}
        userId={userId}
        userRatings={userRatings}
      />)
      expect(wrapper).toMatchSnapshot();
    });

    it('should match the snapshot when the user is logged in and the movie has been rated', () => {
      const isLoggedIn = true
      wrapper = shallow(<MoviePage 
        id={15}
        movies={movies}
        isLoggedIn={isLoggedIn}
        userId={userId}
        userRatings={userRatings}
      />)
      expect(wrapper).toMatchSnapshot();
    });

    it('should match the snapshot when the isLoggedIn is false', () => {
      const isLoggedIn = false
      wrapper = shallow(<MoviePage 
        id={15}
        movies={movies}
        isLoggedIn={isLoggedIn}
        userId={userId}
        userRatings={userRatings}
      />)
      expect(wrapper).toMatchSnapshot();
    });

    it('should match the snapshot when the user is logged in and the correct movie has been rated', () => {
      const isLoggedIn = true
      wrapper = shallow(<MoviePage 
        id={18}
        movies={movies}
        isLoggedIn={isLoggedIn}
        userId={userId}
        userRatings={userRatings}
      />)
      expect(wrapper).toMatchSnapshot();
    });

    it('should match the snapshot when the user is logged in and the correct movie has been rated', () => {
      const isLoggedIn = true
      wrapper = shallow(<MoviePage 
        id={16}
        movies={movies}
        isLoggedIn={isLoggedIn}
        userId={userId}
        userRatings={userRatings}
      />)
      expect(wrapper).toMatchSnapshot();
    });
    describe('handleRatingSubmit', () => {
      it('should be called when a user clicks on the rate button', async () => {
        const apiCalls = require('../../apiCalls');
        const isLoggedIn = true;
        const wrapper = shallow(<MoviePage 
          id={2}
          movies={movies}
          isLoggedIn={isLoggedIn}
          userId={userId}
          userRatings={userRatings}
        />);
        const postSpy = jest.spyOn(apiCalls, 'postUserRating');
        const ratingsSpy = jest.spyOn(apiCalls, 'fetchRatings');

        postSpy.mockResolvedValue();
        ratingsSpy.mockResolvedValue();

        wrapper.find('.rate-btn').simulate('click');

        await Promise.resolve();

        expect(postSpy).toHaveBeenCalledTimes(1);
        expect(ratingsSpy).toHaveBeenCalledTimes(1);
      });
    });

    describe('deleteRating', () => {
      it('should invoke deleteApiRating when the delete rating button is clicked', async () => {
        const apiCalls = require('../../apiCalls');
        const isLoggedIn = true;
        const wrapper = shallow(<MoviePage
          id={18}
          movies={movies}
          isLoggedIn={isLoggedIn}
          userId={userId}
          userRatings={userRatings}
        />);
        const deleteSpy = jest.spyOn(apiCalls, 'deleteApiRating');
        
        deleteSpy.mockResolvedValue();

        wrapper.find('.delete-rating-btn').simulate('click');

        await Promise.resolve();

        expect(deleteSpy).toHaveBeenCalledTimes(1);
      });

      it('should invoke deleteRating when the delete rating button is clicked', () => {
        const isLoggedIn = true;
        const wrapper = shallow(<MoviePage 
          id={18}
          movies={movies}
          isLoggedIn={isLoggedIn}
          userId={userId}
          userRatings={userRatings}
        />)
        wrapper.instance().deleteRating = jest.fn();

        wrapper.find('.delete-rating-btn').simulate('click');

        expect(wrapper.instance().deleteRating).toHaveBeenCalled()
      })
    });

    describe('determineIcon', () => {
      let images, isLoggedIn, wrapper;

      beforeEach(() => {
        images = require('../../images/images');
        isLoggedIn = true;
        wrapper = shallow(<MoviePage 
          id={2}
          movies={movies}
          isLoggedIn={isLoggedIn}
          userId={userId}
          userRatings={userRatings}
        />);
      });

      it('should return a good Tomatillo image if the rating is higher than 6', () => {
        const expected = "goodTomatillo.png";
        expect(wrapper.instance().determineIcon(9)).toEqual(expected);
      });

      it('should return a good Tomatillo image if the rating is higher than 6', () => {
        const expected = "okayTomatillo.png";
        expect(wrapper.instance().determineIcon(5)).toEqual(expected);
      });

      it('should return a bad Tomatillo image if the rating is higher than 6', () => {
        const expected = "badTomatillo.png";
        expect(wrapper.instance().determineIcon(3)).toEqual(expected);
      });
    });

    describe('onStarClick', () => {
      it('should update state when onStarClick is invoked', () => {
        wrapper.instance().setState({rating: 1});
        wrapper.instance().onStarClick(7);
        const expected = {rating: 7};
        expect(wrapper.state()).toEqual(expected);
      });
    });

  });
  describe('mapDispatchToProps', () => {
    it('should call dispatch with updateUserRatings', () => {
      const userRatings = [
        {
          id:22,
          user_id:6,
          movie_id:4,
          rating:10,
          created_at:"2019-12-19T21:47:31.647Z",
          updated_at:"2019-12-19T21:47:31.647Z"
        },
        {
          id:23,
          user_id:6,
          movie_id:15,
          rating:8,
          created_at:"2019-12-19T22:05:16.098Z",
          updated_at:"2019-12-19T22:05:16.098Z"
        },
        {
          id:16,
          user_id:6,
          movie_id:16,
          rating:5,
          created_at:"2019-12-19T22:05:16.098Z",
          updated_at:"2019-12-19T22:05:16.098Z"
        },
        {
          id:18,
          user_id:6,
          movie_id:18,
          rating:2,
          created_at:"2019-12-19T22:05:16.098Z",
          updated_at:"2019-12-19T22:05:16.098Z"
        }
      ];
      const mockDispatch = jest.fn();
      const actionToDispatch = updateUserRatings(userRatings);
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.updateUserRatings(userRatings);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
  describe('mapStateToProps', () => {
    it('should return a props object with movies object, isLoggedIn, userId, and userRatings', () => {
      const mockMovies = {movies: 
        [{
          id:2,
          title:"Jumanji: The Next Level",
          poster_path:"https://image.tmdb.org/t/p/original//l4iknLOenijaB85Zyb5SxH1gGz8.jpg",
          backdrop_path:"https://image.tmdb.org/t/p/original//zTxHf9iIOCqRbxvl8W5QYKrsMLq.jpg",
          release_date:"2019-12-04",
          overview:"In Jumanji: The Next Level, the gang is back to escape the world's most dangerous game.",
          average_rating:6.166666666666667
        },
        {
          id:15,
          title:"Ad Astra",
          poster_path:"https://image.tmdb.org/t/p/original//xBHvZcjRiWyobQ9kxBhO6B2dtRI.jpg",
          backdrop_path:"https://image.tmdb.org/t/p/original//5BwqwxMEjeFtdknRV792Svo0K1v.jpg",
          release_date:"2019-09-17",
          overview:"The near future, astronaut Roy McBride undertakes a mission in search of the unknown.",
          average_rating:5
        },
        {
          id:16,
          title:"Star Wars: The Rise Of Skywalker",
          poster_path:"https://image.tmdb.org/t/p/original//xBHvZcjRiWyobQ9kxBhO6B2dtRI.jpg",
          backdrop_path:"https://image.tmdb.org/t/p/original//5BwqwxMEjeFtdknRV792Svo0K1v.jpg",
          release_date:"2019-09-17",
          overview:"The conclusion of the Skywalker timeline, in a galaxy far far away.",
          average_rating:5
        },
      
        {
          id:18,
          title:"The Count of Monte Cristo",
          poster_path:"https://image.tmdb.org/t/p/original//xBHvZcjRiWyobQ9kxBhO6B2dtRI.jpg",
          backdrop_path:"https://image.tmdb.org/t/p/original//5BwqwxMEjeFtdknRV792Svo0K1v.jpg",
          release_date:"2019-09-17",
          overview:"The ultimate tale of revenge and adventure.",
          average_rating:10
        }]
      };
      const mockUserRatings = [
        {
          id:22,
          user_id:6,
          movie_id:4,
          rating:10,
          created_at:"2019-12-19T21:47:31.647Z",
          updated_at:"2019-12-19T21:47:31.647Z"
        },
        {
          id:23,
          user_id:6,
          movie_id:15,
          rating:8,
          created_at:"2019-12-19T22:05:16.098Z",
          updated_at:"2019-12-19T22:05:16.098Z"
        },
        {
          id:16,
          user_id:6,
          movie_id:16,
          rating:5,
          created_at:"2019-12-19T22:05:16.098Z",
          updated_at:"2019-12-19T22:05:16.098Z"
        },
        {
          id:18,
          user_id:6,
          movie_id:18,
          rating:2,
          created_at:"2019-12-19T22:05:16.098Z",
          updated_at:"2019-12-19T22:05:16.098Z"
        }
      ];

      const mockState = {
        movies: mockMovies,
        isLoggedIn: true,
        userId: 6,
        userRatings: mockUserRatings,
        test: {title: 'Object to show that only requested props are being mapped'}
      };

      const expected = {
        movies: mockMovies,
        isLoggedIn: true,
        userId: 6,
        userRatings: mockUserRatings
      };

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });
});