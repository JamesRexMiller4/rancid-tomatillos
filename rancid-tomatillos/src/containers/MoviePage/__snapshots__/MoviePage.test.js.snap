// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Page Movie Page Component should match the snapshot when the isLoggedIn is false 1`] = `
<main
  className="movie-page-main"
>
  <section
    className="movie-page-section"
    style={
      Object {
        "backgroundImage": "url(https://image.tmdb.org/t/p/original//5BwqwxMEjeFtdknRV792Svo0K1v.jpg)",
      }
    }
  >
    <article
      className="movie-page-poster"
      style={
        Object {
          "backgroundImage": "url(https://image.tmdb.org/t/p/original//xBHvZcjRiWyobQ9kxBhO6B2dtRI.jpg)",
        }
      }
    />
    <section
      className="movie-info-container"
    >
      <article
        className="movie-page-details"
      >
        <h2>
          Ad Astra
        </h2>
        <p
          className="movie-overview"
        >
          The near future, astronaut Roy McBride undertakes a mission in search of the unknown.
        </p>
        <h3>
          Release Date
        </h3>
        <p>
          2019-09-17
        </p>
      </article>
      <article
        className="movie-page-ratings"
      >
        <section
          className="average-rating-container"
        >
          <h2>
            Average Rating
          </h2>
          <div
            className="rating-section"
          >
            <img
              alt="rancid tomatillo"
              className="tomatillo-icon"
              src="okayTomatillo.png"
            />
            <h2>
              5
            </h2>
          </div>
        </section>
      </article>
    </section>
  </section>
</main>
`;

exports[`Movie Page Movie Page Component should match the snapshot when the user is logged in and the correct movie has been rated 1`] = `
<main
  className="movie-page-main"
>
  <section
    className="movie-page-section"
    style={
      Object {
        "backgroundImage": "url(https://image.tmdb.org/t/p/original//5BwqwxMEjeFtdknRV792Svo0K1v.jpg)",
      }
    }
  >
    <article
      className="movie-page-poster"
      style={
        Object {
          "backgroundImage": "url(https://image.tmdb.org/t/p/original//xBHvZcjRiWyobQ9kxBhO6B2dtRI.jpg)",
        }
      }
    />
    <section
      className="movie-info-container"
    >
      <article
        className="movie-page-details"
      >
        <h2>
          The Count of Monte Cristo
        </h2>
        <p
          className="movie-overview"
        >
          The ultimate tale of revenge and adventure.
        </p>
        <h3>
          Release Date
        </h3>
        <p>
          2019-09-17
        </p>
      </article>
      <article
        className="movie-page-ratings"
      >
        <section
          className="average-rating-container"
        >
          <h2>
            Average Rating
          </h2>
          <div
            className="rating-section"
          >
            <img
              alt="rancid tomatillo"
              className="tomatillo-icon"
              src="goodTomatillo.png"
            />
            <h2>
              10
            </h2>
          </div>
        </section>
        <section
          className="average-rating-container"
        >
          <h2>
            User Rating
          </h2>
          <div
            className="rating-section"
          >
            <img
              alt="rancid tomatillo"
              className="tomatillo-icon"
              src="badTomatillo.png"
            />
            <h2>
              2
            </h2>
            <button
              className="delete-rating-btn"
              onClick={[Function]}
              type="button"
            >
              Delete Rating
            </button>
          </div>
        </section>
      </article>
    </section>
  </section>
</main>
`;

exports[`Movie Page Movie Page Component should match the snapshot when the user is logged in and the correct movie has been rated 2`] = `
<main
  className="movie-page-main"
>
  <section
    className="movie-page-section"
    style={
      Object {
        "backgroundImage": "url(https://image.tmdb.org/t/p/original//5BwqwxMEjeFtdknRV792Svo0K1v.jpg)",
      }
    }
  >
    <article
      className="movie-page-poster"
      style={
        Object {
          "backgroundImage": "url(https://image.tmdb.org/t/p/original//xBHvZcjRiWyobQ9kxBhO6B2dtRI.jpg)",
        }
      }
    />
    <section
      className="movie-info-container"
    >
      <article
        className="movie-page-details"
      >
        <h2>
          Star Wars: The Rise Of Skywalker
        </h2>
        <p
          className="movie-overview"
        >
          The conclusion of the Skywalker timeline, in a galaxy far far away.
        </p>
        <h3>
          Release Date
        </h3>
        <p>
          2019-09-17
        </p>
      </article>
      <article
        className="movie-page-ratings"
      >
        <section
          className="average-rating-container"
        >
          <h2>
            Average Rating
          </h2>
          <div
            className="rating-section"
          >
            <img
              alt="rancid tomatillo"
              className="tomatillo-icon"
              src="okayTomatillo.png"
            />
            <h2>
              5
            </h2>
          </div>
        </section>
        <section
          className="average-rating-container"
        >
          <h2>
            User Rating
          </h2>
          <div
            className="rating-section"
          >
            <img
              alt="rancid tomatillo"
              className="tomatillo-icon"
              src="okayTomatillo.png"
            />
            <h2>
              5
            </h2>
            <button
              className="delete-rating-btn"
              onClick={[Function]}
              type="button"
            >
              Delete Rating
            </button>
          </div>
        </section>
      </article>
    </section>
  </section>
</main>
`;

exports[`Movie Page Movie Page Component should match the snapshot when the user is logged in and the movie has been rated 1`] = `
<main
  className="movie-page-main"
>
  <section
    className="movie-page-section"
    style={
      Object {
        "backgroundImage": "url(https://image.tmdb.org/t/p/original//5BwqwxMEjeFtdknRV792Svo0K1v.jpg)",
      }
    }
  >
    <article
      className="movie-page-poster"
      style={
        Object {
          "backgroundImage": "url(https://image.tmdb.org/t/p/original//xBHvZcjRiWyobQ9kxBhO6B2dtRI.jpg)",
        }
      }
    />
    <section
      className="movie-info-container"
    >
      <article
        className="movie-page-details"
      >
        <h2>
          Ad Astra
        </h2>
        <p
          className="movie-overview"
        >
          The near future, astronaut Roy McBride undertakes a mission in search of the unknown.
        </p>
        <h3>
          Release Date
        </h3>
        <p>
          2019-09-17
        </p>
      </article>
      <article
        className="movie-page-ratings"
      >
        <section
          className="average-rating-container"
        >
          <h2>
            Average Rating
          </h2>
          <div
            className="rating-section"
          >
            <img
              alt="rancid tomatillo"
              className="tomatillo-icon"
              src="okayTomatillo.png"
            />
            <h2>
              5
            </h2>
          </div>
        </section>
        <section
          className="average-rating-container"
        >
          <h2>
            User Rating
          </h2>
          <div
            className="rating-section"
          >
            <img
              alt="rancid tomatillo"
              className="tomatillo-icon"
              src="goodTomatillo.png"
            />
            <h2>
              8
            </h2>
            <button
              className="delete-rating-btn"
              onClick={[Function]}
              type="button"
            >
              Delete Rating
            </button>
          </div>
        </section>
      </article>
    </section>
  </section>
</main>
`;

exports[`Movie Page Movie Page Component should match the snapshot when the user is logged in and the movie is not rated 1`] = `
<main
  className="movie-page-main"
>
  <section
    className="movie-page-section"
    style={
      Object {
        "backgroundImage": "url(https://image.tmdb.org/t/p/original//zTxHf9iIOCqRbxvl8W5QYKrsMLq.jpg)",
      }
    }
  >
    <article
      className="movie-page-poster"
      style={
        Object {
          "backgroundImage": "url(https://image.tmdb.org/t/p/original//l4iknLOenijaB85Zyb5SxH1gGz8.jpg)",
        }
      }
    />
    <section
      className="movie-info-container"
    >
      <article
        className="movie-page-details"
      >
        <h2>
          Jumanji: The Next Level
        </h2>
        <p
          className="movie-overview"
        >
          In Jumanji: The Next Level, the gang is back to escape the world's most dangerous game.
        </p>
        <h3>
          Release Date
        </h3>
        <p>
          2019-12-04
        </p>
      </article>
      <article
        className="movie-page-ratings"
      >
        <section
          className="average-rating-container"
        >
          <h2>
            Average Rating
          </h2>
          <div
            className="rating-section"
          >
            <img
              alt="rancid tomatillo"
              className="tomatillo-icon"
              src="goodTomatillo.png"
            />
            <h2>
              7
            </h2>
          </div>
        </section>
        <section
          className="user-rating-container"
        >
          <h2>
            Rate This Movie
          </h2>
          <section>
            <StarRatingComponent
              editing={true}
              emptyStarColor="#333"
              name="rating"
              onStarClick={[Function]}
              starColor="#ffb400"
              starCount={10}
              value={1}
            />
          </section>
          <button
            className="rate-btn"
            onClick={[Function]}
            type="button"
          >
            <img
              alt="image of rancid tomatillo"
              className="tomatillo-icon"
              src="badTomatillo.png"
            />
            1
             Rate It
          </button>
        </section>
      </article>
    </section>
  </section>
</main>
`;
